day01/day01_v2.py

```diff
@@ -0,0 +1,37 @@
+
+from typing import List;
+import os;
+import time;
+import numpy as np
+
+def read_input(file_name):
+    current_path = os.path.dirname(__file__)
+    lines : List[int] = []
+    try:
+        with open(os.path.join(current_path, file_name), 'r') as file:
+            for line in file:
+                lines.append(int(line.strip()))  
+    except Exception as e:
+        print(f"An error occurred: {e}")
+    return lines
+
+def freq_calculations(inputs):
+    cumlsum_inputs = np.cumsum(inputs)
```

Comment:
using numpy here is overall no bad idea, nevertheless I think you tricked yourself. I had to lookup the result of cumsum myself.